pages/_app.js

```diff
@@ -79,8 +79,22 @@ export default function App({ Component, pageProps }) {
       id: uid(),
     };
 
-    setCollections([newCollection, ...collections]);
-    event.target.reset();
+    if (!collectionToUpdate_id) {
+      setCollections([newCollection, ...collections]);
+      event.target.reset();
+    } else {
+      setCollections(
+        collections.map((collection) =>
+          collection.id === collectionToUpdate_id
+            ? {
+                id: collectionToUpdate_id,
+                ...data,
+              }
+            : collection
+        )
+      );
+      router.back();
+    }
```

Comment:
please create 2 functions instead, one for creating a collection, one for updating. This makes the code simpler and easier to maintain.